crates/pico/tests/shared_dependency.rs

```diff
@@ -0,0 +1,67 @@
+use calc::{ast::Program, error::Result, eval::eval, lexer::Lexer, parser::Parser};
+use pico::storage::DefaultStorage;
+use pico_core::{database::Database, source::SourceId};
+use pico_macros::{memo, Db, Source};
+
+mod calc;
+
+/// Assert that we correctly compare epochs when 2 queries share the same dependency,
+/// i.e. use the same memoized `parse_ast` function
+#[test]
+fn shared_dependency() {
+    let mut state = TestDatabase::default();
+
+    let input = state.set(Input {
+        key: "input",
+        value: "2 + 2 * 2".to_string(),
+    });
+
+    let result = evaluate(&mut state, input);
```

Comment:
can we add the same `SUM_COUNTER`-type stuff to assert that we're calling the functions the expected number of times?